src/vs/workbench/services/runtimeStartup/common/runtimeStartup.ts

```diff
@@ -1395,6 +1395,9 @@ export class RuntimeStartupService extends Disposable implements IRuntimeStartup
 		// `exit_code` was `0`, so we don't treat it as a crash. If the `exit_code` had not been
 		// `0`, then `onKernelExited()` would have upgraded the crash from `Unknown` to `Error`.
 		if (exit.reason !== RuntimeExitReason.Error) {
+			if (exit.reason !== RuntimeExitReason.Restart) {
```

Comment:
Minor nit, would it be simpler if we centralized cleaning up and extracted it to the `onDidEndSession` listener? E.g. if `restartAfterCrash` returned a value indicating whether it restarted and we used that to decide whether to cleanup, or if we checked whether a restart would happen in the `onDidEndSession` listener and cleaned up in an `else` clause